mailpoet/assets/js/src/notices/bridge-ping-error-notice.tsx

```diff
@@ -0,0 +1,85 @@
+import { useState, useEffect } from 'react';
+import { extractPageNameFromUrl } from 'common/functions';
+import { Notice } from 'notices/notice';
+import { MailPoet } from 'mailpoet';
+import ReactStringReplace from 'react-string-replace';
+
+const LOCAL_STORAGE_KEY = 'mailpoet_mss_ping_last_success_at';
+const CACHE_TIME = 60 * 60 * 1000; // 1 hour
+
+const cacheSuccessfulResult = () => {
+  window.localStorage.setItem(LOCAL_STORAGE_KEY, Date.now().toString());
+};
+
+const isSuccessfulResultCached = () => {
+  const lastSuccessAt = window.localStorage.getItem(LOCAL_STORAGE_KEY);
+  return (
+    lastSuccessAt && Date.now() - parseInt(lastSuccessAt, 10) <= CACHE_TIME
+  );
+};
+
+function BridgePingErrorNotice() {
+  const [bridgeError, setBridgeError] = useState<string | null>(null);
+
+  useEffect(() => {
+    // Show only when sending method is MailPoet
+    if (MailPoet.mtaMethod !== 'MailPoet') return;
+    // Do not show on the MailPoet Help page, it has its own MSS status indication
+    if (extractPageNameFromUrl() === 'help') return;
+
+    // Don't make a request if we have a recently cached successful result
+    if (isSuccessfulResultCached()) return;
+
+    void MailPoet.Ajax.post({
+      api_version: MailPoet.apiVersion,
+      endpoint: 'Services',
+      action: 'pingBridge',
+    })
+      .done(() => {
+        cacheSuccessfulResult();
+        // Do not display the notice if there are no errors
+        setBridgeError(null);
+      })
+      .fail((errorResponse) => {
+        const errObject = errorResponse?.errors?.[0];
+        if (errObject.error && errObject.message) {
+          // Display MSS error if server responded with error message
+          setBridgeError(errObject.message);
+        } else {
+          // Do not display the notice if there is no server response
+          setBridgeError(null);
+        }
+      });
+  }, []);
```

Comment:
This piece of code, as it is pretty similar to the code for the cron ping, is a nice candidate for a custom hook. Something like `const isError  = useServiceCheck('pingBridge')`. It could encapsulate also the caching code. I consider the change nice to have.

Reply:
I agree that this code is very similar to the cron ping code and can be extracted to a separate hook, but I don't see any more similar notices coming in the future, so this hook will likely be used only in two places. Seems a bit like premature optimization. I can still do it, though, if you insist

Reply:
@costasovo I've decided to incorporate your suggestion and extract the common logic into a custom hook. Please check.